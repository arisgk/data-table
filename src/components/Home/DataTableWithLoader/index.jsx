import DataTableWithLoader from './DataTableWithLoader';

export default DataTableWithLoader;
