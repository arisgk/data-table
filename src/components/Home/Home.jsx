import React from 'react';
import AppBar from 'material-ui/AppBar';

const styles = {
  container: {
    height: '100vh',
    display: 'flex',
    flexDirection: 'column',
  },
};

const Home = () => (
  <div style={styles.container}>
    <AppBar title="Title" />
    <div>Hello World</div>
  </div>
);

export default Home;
