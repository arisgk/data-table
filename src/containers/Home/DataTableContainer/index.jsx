import { gql, graphql, compose } from 'react-apollo';
import { connect } from 'react-redux';
import { startAdding, cancelUserDialog, select, clearSelection } from '../../../actions/home';
import DataTableWithLoader from '../../../components/Home/DataTableWithLoader';

const listQuery = gql`
  query UserListQuery {
    users {
      id
      firstName
      lastName
      age
    }
  }
`;

const createMutation = gql`
  mutation addUser($firstName: String!, $lastName: String!, $age: Int) {
    addUser(firstName: $firstName, lastName: $lastName, age: $age) {
      id
      firstName
      lastName
      age
    }
  }
`;

const options = {
  props: ({ ownProps, ...dataProps }) => ({
    ...ownProps,
    ...dataProps,
  }),
};

const mapStateToProps = state => ({
  adding: state.home.ui.adding,
  listQuery,
});

const mapDispatchToProps = dispatch => ({
  onStartAdding: () => dispatch(startAdding()),
  onCancelUserDialog: () => dispatch(cancelUserDialog()),
  onTableRowSelection: (rows, entities) => {
    if (rows && rows.length > 0 && entities) {
      if (rows === 'all') {
        dispatch(select(entities.map(entity => entity.id)));
      } else if (rows === 'none') {
        dispatch(clearSelection());
      } else {
        dispatch(select(rows.map(row => entities[row].id)));
      }
    } else {
      dispatch(clearSelection());
    }
  },
});

const DataTableWithDataAndState = compose(
  graphql(listQuery, options),
  graphql(createMutation, options),
  connect(mapStateToProps, mapDispatchToProps),
)(DataTableWithLoader);

export default DataTableWithDataAndState;
