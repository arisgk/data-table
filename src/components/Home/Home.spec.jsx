/* global describe, it, expect */
import React from 'react';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';
import Home from './Home';

describe('Home', () => {
  it('renders correctly', () => {
    const tree = renderer.create(
      <MuiThemeProvider>
        <Home />
      </MuiThemeProvider>,
    );

    expect(tree).toMatchSnapshot();
  });
});
