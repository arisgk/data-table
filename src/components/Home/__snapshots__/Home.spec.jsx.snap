// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home renders correctly 1`] = `
<div
  style={
    Object {
      "display": "flex",
      "flexDirection": "columb",
      "height": "100vh",
    }
  }
>
  <div
    className={undefined}
    style={
      Object {
        "WebkitTapHighlightColor": "rgba(0,0,0,0)",
        "backgroundColor": "#00bcd4",
        "borderRadius": "0px",
        "boxShadow": "0 1px 6px rgba(0, 0, 0, 0.12),
               0 1px 4px rgba(0, 0, 0, 0.12)",
        "boxSizing": "border-box",
        "color": "rgba(0, 0, 0, 0.87)",
        "display": "flex",
        "fontFamily": "Roboto, sans-serif",
        "muiPrepared": true,
        "paddingLeft": 24,
        "paddingRight": 24,
        "position": "relative",
        "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
        "width": "100%",
        "zIndex": 1100,
      }
    }
  >
    <button
      disabled={false}
      href={undefined}
      onBlur={[Function]}
      onClick={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onKeyUp={[Function]}
      onMouseEnter={[Function]}
      onMouseLeave={[Function]}
      onMouseOut={[Function]}
      onTouchStart={[Function]}
      style={
        Object {
          "WebkitTapHighlightColor": "rgba(0, 0, 0, 0)",
          "background": "none",
          "border": 10,
          "boxSizing": "border-box",
          "cursor": "pointer",
          "display": "inline-block",
          "fontFamily": "Roboto, sans-serif",
          "fontSize": 0,
          "fontWeight": "inherit",
          "height": 48,
          "margin": 0,
          "marginLeft": -16,
          "marginRight": 8,
          "marginTop": 8,
          "muiPrepared": true,
          "outline": "none",
          "overflow": "visible",
          "padding": 12,
          "position": "relative",
          "textDecoration": "none",
          "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
          "verticalAlign": null,
          "width": 48,
          "zIndex": 1,
        }
      }
      tabIndex={0}
      type="button"
    >
      <div
        onMouseDown={[Function]}
        onMouseLeave={[Function]}
        onMouseUp={[Function]}
        onTouchEnd={[Function]}
        onTouchStart={[Function]}
      >
        <svg
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          style={
            Object {
              "color": "#ffffff",
              "display": "inline-block",
              "fill": "#ffffff",
              "height": 24,
              "muiPrepared": true,
              "transition": "all 450ms cubic-bezier(0.23, 1, 0.32, 1) 0ms",
              "userSelect": "none",
              "width": 24,
            }
          }
          viewBox="0 0 24 24"
        >
          <path
            d="M3 18h18v-2H3v2zm0-5h18v-2H3v2zm0-7v2h18V6H3z"
          />
        </svg>
      </div>
    </button>
    <h1
      onClick={[Function]}
      style={
        Object {
          "boxFlex": 1,
          "color": "#ffffff",
          "flex": "1",
          "fontSize": 24,
          "fontWeight": 400,
          "height": 64,
          "letterSpacing": 0,
          "lineHeight": "64px",
          "margin": 0,
          "muiPrepared": true,
          "overflow": "hidden",
          "paddingTop": 0,
          "textOverflow": "ellipsis",
          "whiteSpace": "nowrap",
        }
      }
    >
      Title
    </h1>
  </div>
  <div>
    Hello World
  </div>
</div>
`;
